stdlib/public/core/Identifiable.swift

```diff
@@ -10,11 +10,33 @@
 //
 //===----------------------------------------------------------------------===//
 
-/// A class of types whose instances hold the value of an entity with stable identity.
+/// A class of types whose instances hold the value of an entity with stable
+/// identity.
+///
+/// `Identifiable` leaves the duration and scope of the identity unspecified.
+/// For example, identities could be:
```

Comment:
The abstract above says "what" this is, so it's always ideal to give some kind of "why/when" statement at the beginning of the discussion. Maybe something like:

> `User` type 
```suggestion
/// Use the `Identifiable` protocol to provide a stable notion of identity to a 
/// class or value type. For example, you could define a `User` type with an `id`
/// property that is stable across your app and your app's database storage. 
/// You could use the `id` property to identify a particular user even if other 
/// data field's change, such as the user's name.
///
/// `Identifiable` leaves the duration and scope of the identity unspecified.
/// Identities could be any of the following:
```